### ISSUE TITLE
DatePicker: Mobile date is parsed incorrectly when changing year

### ISSUE THREAD

**Issue by nchutchind** [buefy/buefy#1397]
### Using v-model with the mobile native date picker doesn't parse dates correctly.

**Buefy** version: 0.7.5
**Vuejs** version: 2.6.10
**OS/Browser**: Windows

### Description

All date inputs format dates as YYYY-MM-DD, but your locale may render the input with a format such as mm/dd/yyyy. When using v-model and the mobile-native datepicker, the datepicker attempts to format the date on each keystroke. This can lead to some utterly nuts looking date formats. For example, if you want to enter in the date 06/04/1992, when you enter the 1 for the year, the date being parsed at that moment is '0001-06-04', which gets parsed to '01/06/2004'. '11/15/2019' would get parsed as '0002-11-15' and end up as '02/11/2015'.

### Expected behavior

I expected the date to not jump around.

### Actual behavior

The date jumped all over the place.

**Comment by jtommy** [comment]
May you're right but I don't understand how can you edit the date on mobile?

**Comment by nchutchind** [comment]
@jtommy Honestly, it's probably not a huge issue. The native datepicker works on an actual mobile device, but it doesn't work in the emulated mobile devices in Chrome. I ended up working around the issue by looking at navigator.platform to see if we're on an actual mobile device or on a Windows or Mac. If we're on an actual mobile device, then I set mobile-native to true, otherwise false.

This probably doesn't affect most people, but I have users who are using the mobile views of my app in the emulator because they like it that way.
